third_party/WebKit/Source/wtf/text/TextCodecICUAlternatives.cpp

```diff
@@ -0,0 +1,69 @@
+// Copyright 2014 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "config.h"
+#include "wtf/text/TextCodecICUAlternatives.h"
+#include "wtf/PassOwnPtr.h"
+#include "wtf/text/CharacterNames.h"
+#include "wtf/text/WTFString.h"
+#include "base/icu_alternatives_on_android/icu_utils.h"
+
+namespace WTF {
+
+TextCodecICUAlternatives::TextCodecICUAlternatives(const TextEncoding& encoding)
+    : m_encoding(encoding)
+{
+}
+
+void TextCodecICUAlternatives::registerEncodingNames(EncodingNameRegistrar registrar)
+{
+    //TODO, add more charsets supported by Java.
```

Comment:
In Java side, theoretically it can support lots of charset.  This error is caused by decoding Shift-JIS, so I just register Shift-JIS here.  we can add other charset if necessary in future.